armory/datasets/generator.py

```diff
@@ -129,6 +127,9 @@ def __init__(
             # SEE: tfds.as_numpy
             # https://github.com/tensorflow/datasets/blob/v4.7.0/tensorflow_datasets/core/dataset_utils.py#L141-L176
             raise NotImplementedError(f"framework {framework}")
+        
+        if info.metadata is None:
+            info.metadata = tfds.core.MetadataDict()
```

Comment:
this line yields an error when the conditional is met. Try testing with e.g. cifar dataset:

```

 File "/workspace/armory/datasets/generator.py", line 132, in __init__
    if info.metadata is None:
       │    └ <property object at 0x7fabca4d7270>
       └ tfds.core.DatasetInfo(
             name='cifar10',
             full_name='cifar10/3.0.2',
             description="""
             The CIFAR-10 dataset consis...

AttributeError: can't set attribute

```